com.unity.render-pipelines.high-definition/Runtime/Lighting/AtmosphericScattering/Fog.cs

```diff
@@ -51,28 +47,50 @@ public class Fog : VolumeComponent
-
-        /// <summary>Sets the distance (in meters) from the Camera's Near Clipping Plane to the back of the Camera's volumetric lighting buffer.</summary>
-        [Tooltip("Sets the distance (in meters) from the Camera's Near Clipping Plane to the back of the Camera's volumetric lighting buffer.")]
+        /// <summary>Sets the distance (in meters) from the Camera's Near Clipping Plane to the back of the Camera's volumetric lighting buffer. The lower the distance is, the higher the fog quality is.</summary>
         public MinFloatParameter depthExtent = new MinFloatParameter(64.0f, 0.1f);
+        /// <summary> Controls which version of the effect should be used. </summary>
+        [Tooltip("Defines which technique is used for denoising the volumetric fog. Reprojection is very effective for static lighting, but can lead to severe ghosting for highly dynamic lighting. Gaussian is a good alternative for dynamic lighting. Using both techniques can give high quality results but will increase significantly the overall cost of the effect.")]
+        public FogDenoisingModeParameter denoisingMode = new FogDenoisingModeParameter(FogDenoisingMode.Gaussian);
+
+        // Advanced parameters
+        /// <summary>Volumetric fog anisotropy.</summary>
+        public ClampedFloatParameter anisotropy = new ClampedFloatParameter(0.0f, -1.0f, 1.0f);
         /// <summary>Controls the distribution of slices along the Camera's focal axis. 0 is exponential distribution and 1 is linear distribution.</summary>
         [Tooltip("Controls the distribution of slices along the Camera's focal axis. 0 is exponential distribution and 1 is linear distribution.")]
         public ClampedFloatParameter sliceDistributionUniformity = new ClampedFloatParameter(0.75f, 0, 1);
+
+        // Limit parameters for the fog quality
+        internal const float minFogScreenResolutionPercentage = (1.0f / 16.0f) * 100;
+        internal const float optimalFogScreenResolutionPercentage = (1.0f / 8.0f) * 100;
+        internal const float maxFogScreenResolutionPercentage = 0.5f * 100;
+        internal const int maxFogSliceCount = 512;
+
+        /// <summary>Two modes are available for controlling the performance and quality for the volumetric fog. The balance mode allows for an intuitive, performance oriented approach. The manual mode gives access to the internal set of parameters required for the evaluation.</summary>
+        [Tooltip("Two modes are available for controlling the performance and quality for the volumetric fog. The balance mode allows for an intuitive, performance oriented approach. The manual mode gives access to the internal set of parameters required for the evaluation.")]
+        public FogControlParameter fogControlMode = new FogControlParameter(FogControl.Balance);
         /// <summary>Resolution of the volumetric buffer (3D texture) along the X and Y axes relative to the resolution of the frame buffer.</summary>
```

Comment:
Stores the resolution of the volumetric buffer (3D texture) along the x-axis and y-axis relative to the resolution of the frame buffer.